### PR TITLE
Module5:Blog app - Views

### PR DESCRIPTION
**Pull request by derrick1451**
### Requirements
- Implement the design from the sneak peek wireframes.
  - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)
  - Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
  - Hint: In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).
  - Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
  - Make sure that you have used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Highlights
- RESOLVED
- loom video to confirm
- https://www.loom.com/share/755fff75770c46fe9b3e1dab4f4d259f?sid=b6a9f82d-d16b-424e-8cf3-3ace3e00010c

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,7 @@
-<h1>Lists all users</h1>
\ No newline at end of file
+<h1>Lists all users</h1>
+<%@users.each do |user|%>
+<ul>
+<li>User_name:<%=user.Name%> --- Number of posts:<%=user.PostsCounter%>
+user_id: <%=user.id%></li>
+</ul>
+<%end%>
```
Comment (Munsa1) [inline app/views/users/index.html.erb]: - [ ] The users index page should match the given wireframe design, all components in the given wireframe should be added to your design. Kindly implement this.

Your design
image


Given wireframe
image

- [ ] Clicking on each user should take you to their show page, Kindly implement this. Currently nothing happens

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,12 @@
...
+<h2><%=@user_id.Name%></h2>
+<p><%=@user_id.PostsCounter%></p>
+</div>
+<p><%=@user_id.Bio%><p>
+<%@user_id.most_3_recent_posts.each do |post|%>
+<ul>
+<li> post:<%=post.id%> <br> Title:<%=post.Title%> <br> Text:<%=post.Text%> <br>
+comments:<%=post.CommentsCounter%> <br> Likes: <%=post.LikesCounter%></li>
+</ul>
+<%end%>
```
Comment (Munsa1) [inline app/views/users/show.html.erb]: - [ ] The users show page should follow the given wireframe for design as well. Kindly add the see all posts button to your design and all other missing components.

Your design
image


Given design
image


#### Note that all elements in the given wireframes should be implemented

Comment (Munsa1) [inline app/views/users/show.html.erb]: - [ ] Once you have added it, clicking on the **See all Post** link should open the user's post index as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,9 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<h1>Here is a list of posts for a given user</h1>
+<%@posts.each do |post|%>
+<ul>
+<li>post :<%=post.id%> <br> Text :<%=post.Text%> <br> Coments :<%=post.CommentsCounter%> <br> Likes :<%=post.LikesCounter%>  </li>
+</ul>
+<%post.comments.each do |comments|%>
+<li><%=comments.Text%></li>
```
Comment (Munsa1) [inline app/views/posts/index.html.erb]: - [ ] Kindly Use `five_most_recent_comments` method created in the Post model to render 5 most recent comments in the `app/views/posts/index.html.erb` page

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,9 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<h1>Here is a list of posts for a given user</h1>
+<%@posts.each do |post|%>
+<ul>
+<li>post :<%=post.id%> <br> Text :<%=post.Text%> <br> Coments :<%=post.CommentsCounter%> <br> Likes :<%=post.LikesCounter%>  </li>
```
Comment (Munsa1) [inline app/views/posts/index.html.erb]: - [ ] Kindly ensure your design matches the given design. Again here as for everywhere all components in the given design should be added to your design.

Your design
image


Given design
image

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,11 @@
...
+<div class="post">
+  <div class="post-title">
+    <h2>Post:<%= @post.id %> by <%= @post.author.Name %> </h2>
+    <p>Comments: <%= @post.CommentsCounter %>, Likes: <%= @post.LikesCounter %></p>
+  </div>
+  <h3><%= @post.Title %></h3>
+  <h4><%= @post.Text %></h4>
+  <%@post.five_most_recent_comments.each do |comment|%>
+  <p>User:<%=comment.author.Name%> Comment:<%=comment.Text%>  </p>
+  <%end%>
```
Comment (Munsa1) [inline app/views/posts/show.html.erb]: - [ ] The post show page should also follow the given design and should implement all the given elements on the page.

Your design
image

Given frame
image

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -1 +1,11 @@
-<h1>Here is a post for a given id </h1>
\ No newline at end of file
+<div class="post">
+  <div class="post-title">
+    <h2>Post:<%= @post.id %> by <%= @post.author.Name %> </h2>
+    <p>Comments: <%= @post.CommentsCounter %>, Likes: <%= @post.LikesCounter %></p>
+  </div>
+  <h3><%= @post.Title %></h3>
+  <h4><%= @post.Text %></h4>
+  <%@post.five_most_recent_comments.each do |comment|%>
```
Comment (Munsa1) [inline app/views/posts/show.html.erb]: - [ ] The `five_most_recent_comments` method should not be used here on the show page, here we want to show the post with all the comments, we should use that method on the post index page. Kindly do so

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @derrick1451, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Implemented Views :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

### Take Note :warning: 
- [ ] The biggest requirement is the design of your views, kindly follow the given wireframe design for a basic UI as what we have now is no design at all
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @derrick1451, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Implemented Most Of The Requested Changes :heavy_check_mark: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Munsa1** [comment]
- [ ] Clicking on each user should take the user to the user index page, but currently, there is an error, kindly fix it.

User index
image

After clicking on each user
image

image

## To the next reviewer
- [ ] Since I have no access to the user show page I am unable to determine if these two issues were resolved. 
- [issue 1](https://github.com/derrick1451/Blog-app/pull/6#discussion_r1319144230)
- [issue 2](https://github.com/derrick1451/Blog-app/pull/6#discussion_r1319145543)

**Comment by derrick1451** [comment]
@Munsa1 Yeah l saw that error as well but l guess you haven't pull my current work because when you check my pull request you could see that l actually made a new commit and l pushed it but l guess you had already started reviewing

**Comment by derrick1451** [comment]
dear reviewer this  is the video confirmation
- https://www.loom.com/share/755fff75770c46fe9b3e1dab4f4d259f?sid=b6a9f82d-d16b-424e-8cf3-3ace3e00010c

**Comment by Munsa1** [comment]
> @Munsa1 Yeah l saw that error as well but l guess you haven't pull my current work because when you check my pull request you could see that l actually made a new commit and l pushed it but l guess you had already started reviewing

@derrick1451 understood but please for the next projects please DO not push on a PR once you have submitted it for review, it brings up this exact kind of confusion

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi  @derrick1451 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Good Job :+1:
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] Keep doing the good work :+1: 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**